src/main/java/com/iota/iri/Snapshot.java

```diff
@@ -2,175 +2,177 @@
-        InputStream inputStream = Snapshot.class.getResourceAsStream(snapshotPath);
-        //if resource doesn't exist, read from file system
-        if (inputStream == null) {
-            inputStream = new FileInputStream(snapshotPath);
-        }
-
-        return inputStream;
     }
 
     private static void checkInitialSnapshotIsConsistent(Map<Hash, Long> initialState) {
         if (!isConsistent(initialState)) {
-            log.error("Initial Snapshot inconsistent.");
-            System.exit(-1);
+            throw new IllegalStateException("Initial Snapshot inconsistent.");
         }
     }
 
     private static void checkStateHasCorrectSupply(Map<Hash, Long> initialState) {
         long stateValue = initialState.values().stream().reduce(Math::addExact).orElse(Long.MAX_VALUE);
         if (stateValue != TransactionViewModel.SUPPLY) {
-            log.error("Transaction resolves to incorrect ledger balance: {}", TransactionViewModel.SUPPLY - stateValue);
-            System.exit(-1);
+            throw new IllegalStateException("Transaction resolves to incorrect ledger balance: " + (TransactionViewModel.SUPPLY - stateValue));
         }
     }
 
     private static Map<Hash, Long> initInitialState(String snapshotFile) {
-        String line;
-        Map<Hash, Long> state = new HashMap<>();
-        BufferedReader reader = null;
-        try {
-            InputStream snapshotStream = getSnapshotStream(snapshotFile);
-            BufferedInputStream bufferedInputStream = new BufferedInputStream(snapshotStream);
-            reader = new BufferedReader(new InputStreamReader(bufferedInputStream));
-            while ((line = reader.readLine()) != null) {
-                String[] parts = line.split(";", 2);
-                if (parts.length >= 2) {
-                    String key = parts[0];
-                    String value = parts[1];
+
+        try (InputStream inputStream = Snapshot.class.getResourceAsStream(snapshotFile);
+             BufferedReader reader = new BufferedReader((inputStream == null)
+                 ? new FileReader(snapshotFile) : new InputStreamReader(inputStream))) {
+
+            Map<Hash, Long> state = new HashMap<>();
+
+            reader.lines().forEach(line -> {
+                int index = line.indexOf(';');
+                if (index != -1 && index != line.length() - 1) {
+                    String key = line.substring(0, index);
+                    String value = line.substring(index + 1);
                     state.put(new Hash(key), Long.valueOf(value));
                 }
-            }
+            });
+            return state;
         } catch (IOException e) {
-            //syso is left until logback is fixed
-            System.out.println("Failed to load snapshot.");
-            log.error("Failed to load snapshot.", e);
-            System.exit(-1);
-        }
-        finally {
-            IOUtils.closeQuietly(reader);
```

Comment:
If using try-with-resources, there is no need to close quietly. It will automatically add any extra exceptions as `addSuppressed` in the original exception.